Vagrantfile

```diff
@@ -19,8 +26,8 @@ Vagrant.configure(2) do |config|
     v.customize ["setextradata", :id, "VBoxInternal2/SharedFoldersEnableSymlinksCreate/v-root", "1"]
     v.memory = 2048
   end
-  # General-purpose env var to let scripts know we are in Vagrant.
-  config.vm.provision "shell", inline: 'echo "export VAGRANT=true" >> /etc/profile'
+
+  config.vm.provision "shell", inline: $env_script
```

Comment:
I have moved the env_variable setting to a separate "script", so that it is easier to follow.
Also, `scripts/start.sh` may rely on `is_vagrant_vm` file. This file has to be created before the script is run (as compared to my previous commit).